script/lazytox.py

```diff
@@ -0,0 +1,204 @@
+"""
+import os
+import asyncio
+import sys
+import re
+from collections import namedtuple
+
+RE_ASCII = re.compile(r"\033\[[^m]*m")
+ERROR = namedtuple('ERROR', "file line col msg")
+
+
+def validate_requirements_ok():
+    """Validate requirements, returns True of ok."""
+    # pylint: disable=E0402
+    from .gen_requirements_all import main
+    return main(True) == 0
+
+
+async def read_stream(stream, display):
+    """Read from stream line by line until EOF, display, and capture lines."""
+    output = []
+    while True:
+        line = await stream.readline()
+        if not line:
+            break
+        output.append(line)
+        display(line.decode())  # assume it doesn't block
+    return b''.join(output)
+
+
+async def async_exec(*args, mute_err=False, capture=True):
+    """Execute, return code & log."""
+    try:
+        kwargs = {'loop': LOOP, 'stdout': asyncio.subprocess.PIPE,
+                  'stderr': asyncio.subprocess.STDOUT}
+        if mute_err:
+            kwargs['stderr'] = asyncio.subprocess.DEVNULL
+        if not capture:
+            kwargs['stderr'] = asyncio.subprocess.PIPE
+        # pylint: disable=E1120
+        proc = await asyncio.create_subprocess_exec(*args, **kwargs)
+    except FileNotFoundError as err:
+        print('ERROR: You need to install {}. Could not execute: {}'.format(
+            args[0], ' '.join(args)))
+        raise err
+
+    if capture:
+        # Readin stdout into log
+        stdout, _ = await proc.communicate()
+    else:
+        # read child's stdout/stderr concurrently (capture and display)
+        stdout, _ = await asyncio.gather(
+            read_stream(proc.stdout, sys.stdout.write),
+            read_stream(proc.stderr, sys.stderr.write))
+    exit_code = await proc.wait()
+    # Convert to ASCII
+    stdout = stdout.decode('utf-8')
+    # log = RE_ASCII.sub('', log.decode())
+    return exit_code, stdout
+
+
+async def git():
+    """Exec git."""
+    try:
+        with open('lazytox.log') as file:
+            oldfiles = file.readlines()
+    except FileNotFoundError:
+        oldfiles = []
+
+    try:
+        _, log = await async_exec(
+            'git', 'diff', 'upstream/dev...', '--name-only')
+    except FileNotFoundError:
+        print("Using a cached version of changed files.")
+        return '\n'.join(oldfiles)
+
+    if oldfiles != log:
+        with open('lazytox.log', 'w') as file:
+            file.write(log)
+    return log
+
+
+async def pylint(files):
+    """Exec pylint."""
+    try:
+        # Drops STDERR, it contains info on loading config file
+        _, log = await async_exec('pylint', '-f', 'parseable', *files,
+                                  mute_err=True)
+    except FileNotFoundError:
+        return []
+
+    res = []
+    for line in log.splitlines():
+        line = line.split(':')
+        if len(line) != 4:
+            continue
+        res.append(ERROR(line[0].replace('\\', '/'), line[1], "", line[2]))
```

Comment:
```python
line[0].replace('\\', '/')
```

I think this is for Windows, right? Shouldn't we print out fully valid Windows paths then? Also, you don't replace with `flake8`while you do here...

Reply:
The outputs from `pylint` and `flake8` are not the same. `flake8` was '/'. `pylint` a single `\'. Probably best to do both to be safe. `/` is perfectly valid on Windows if you want to avoid all kinds of escaping